model/task_lifecycle.go

```diff
@@ -242,6 +242,14 @@ func TryResetTask(taskId, user, origin string, detail *apimodels.TaskEndDetail)
 		} else if !t.IsFinished() {
 			if detail != nil {
 				grip.Debugln(msg, "marking as failed")
+				if detail.Status == "" {
```

Comment:
Do we need to to do this again here? MarkEnd calls task.MarkEnd so that log seems like it would be enough.

Reply:
True, wanted to be sure in case of any alternate chains of logic but I suppose we can remove this here.